lib/form/html.js

```diff
@@ -58,14 +58,31 @@ const addRequired = (html, input) => {
         level += `required--${input.required}`;
       }
 
-      // add required class to label
-      let find = new RegExp(`for=['"]\\s*${id}\\s*['"]`);
-      render = render.replace(find, `for="${id}" class="${level}"`);
+      // regex to search for `class` in `label`
+      const reClass = new RegExp('<label[\\w\\s\\W]+?(class=["\'][\\w\\W\\s]+?["\'])[\\w\\s\\W]+?>');
+
+      // regex to search for `for`
+      const reFor = new RegExp('(for=["\'][\\w\\W\\s]+?["\'])');
+
+      // regex to get classes out of `class`
+      const reClasses = new RegExp('class=["\']([\\w\\s\\W]+?)["\']');
+
+      if (Array.isArray(render.match(reClass))) {
+        // first, check if `class` exists in label
+        const cls = render.match(reClass)[1].match(reClasses);
+        const classes = cls[1].trim().split(/\s+/).concat([level]);
+        const newClass = `class="${classes.join(' ')}"`;
+        render = render.replace(cls[0], newClass);
+      }
+      else {
+        // else, we add class after the `for` attribute
+        render = render.replace(reFor, `for="${id}" class="${level}"`);
+      }
```

Comment:
This doesn't look like it will work if there are multiple inputs in the rendered HTML